### PR TITLE
124 check available offers

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Add hooks and component to handle eligibility check for active offers

- [x] - Adds useCheckAvailableOffers hook
- [x] - add pool-criteria matchers
- [x] - add a UI component for checking available offers
- [x] - implement check-avaialble-offers into the segmentation flow
- [x] - add actual criteria

## Additional fixes:
- [x] fix wrong post body for redtent video-upload

About # (link your issue here)
#124 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 export const OffersAgreement = () => {
-  const { updateDataValue } = useContext(WizardContext);
   const { nextStep } = useWizard();
 
   // we need the offersAgreement acceptance for running checkAvailableOffers
-  const handleAccept = () => {
-    updateDataValue("segmentation", "offersAgreement", true);
+  const handleAccept = async () => {
+    // handle through controller
+    await AsyncStorage.setItem("goodid_permission", "true");
```
Comment (sirpy) [inline packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx:16]: what if user doesnt accept?

File: packages/good-design/src/apps/goodid/controllers/SegmentationController.tsx [hunk 2]
```diff
@@ -2,27 +2,53 @@ import React, { useCallback } from "react";
...
 import { isEmpty } from "lodash";
 
-import { SegmentationWizard } from "../wizards/SegmentationWizard";
-import { useCertificatesMap } from "../../../hooks";
+import { SegmentationProps, SegmentationWizard } from "../wizards/SegmentationWizard";
 
-export const SegmentationController = () => {
+const redtentOffer = [
+  {
+    campaign: "RedTent",
```
Comment (sirpy) [inline packages/good-design/src/apps/goodid/controllers/SegmentationController.tsx:20]: what about age?

Comment (L03TJ3) [reply packages/good-design/src/apps/goodid/controllers/SegmentationController.tsx:20]: If I understood the call on tuesday correctly, age was no longer a criteria to match again.

@decentralauren can you re-confirm?

File: packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx (hunk 1 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx:16]: Then we don't have the permission stored in localStorage.
seemed a bit redundant to explicitly store the value and set it to false

Either the permission exists or it doesn't.

**Comment by decentralauren** [comment]
@L03TJ3 confirming that there is no age criteria for the Red Tent offer.

File: packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx (hunk 1 above)
Comment (sirpy) [reply packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx]: if you put false you know user actually went through this stage and actively selected not to share

**Pull request merged**